Improve error message for maven repository location

When the remote repository setting has a type, e.g. `--maven.remote-ropositories.repo1.url=http://repo.spring.io/libs-milestone"` the log message looks like this

```
Caused by: java.lang.IllegalStateException: failed to resolve MavenResource: org.springframework.cloud.task.app:timestamp-task:jar:1.2.0.RC1. Configured remote repository: 
    at org.springframework.cloud.deployer.resource.maven.MavenArtifactResolver.resolve(MavenArtifactResolver.java:285) ~[spring-cloud-deployer-resource-maven-1.1.3.RELEASE.jar!/:1.1.3.RELEASE]
```

It doesn't clearly show that there is an "empty" value for the remote repository.

--- a/spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenArtifactResolver.java
+++ b/spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenArtifactResolver.java
@@ -276,14 +276,14 @@ class MavenArtifactResolver {
 			final StringBuilder errorMessage = new StringBuilder("failed to resolve MavenResource: %s. Configured remote ");
 
 			if (properties.getRemoteRepositories().size() > 1) {
-				errorMessage.append("repositories: ");
+				errorMessage.append("repositories: [");
 			}
 			else {
-				errorMessage.append("repository: ");
+				errorMessage.append("repository: [");
 			}
 
 			errorMessage.append(StringUtils.collectionToCommaDelimitedString(properties.getRemoteRepositories().keySet()));
-
+			errorMessage.append("]");
 			throw new IllegalStateException(
 					String.format(errorMessage.toString(), resource.toString()), e);
 		}

